homeassistant/components/here_weather/weather.py

```diff
@@ -0,0 +1,222 @@
+        )
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return None
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        get_attribute_from_here_data(self.coordinator.data, "humidity")
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        get_attribute_from_here_data(self.coordinator.data, "windSpeed")
+
+    @property
+    def wind_bearing(self):
+        """Return the wind bearing."""
+        get_attribute_from_here_data(self.coordinator.data, "windDirection")
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return None
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        if self.coordinator.data is None:
+            return None
+        data = []
+        for offset in range(len(self.coordinator.data)):
+            data.append(
+                {
+                    ATTR_FORECAST_TIME: get_attribute_from_here_data(
+                        self.coordinator.data, "utcTime", offset
+                    ),
+                    ATTR_FORECAST_TEMP: get_high_or_default_temperature_from_here_data(
+                        self.coordinator.data, self._mode, offset
+                    ),
+                    ATTR_FORECAST_TEMP_LOW: get_low_or_default_temperature_from_here_data(
+                        self.coordinator.data, self._mode, offset
+                    ),
+                    ATTR_FORECAST_PRECIPITATION: calc_precipitation(
+                        self.coordinator.data, offset
+                    ),
+                    ATTR_FORECAST_WIND_SPEED: get_attribute_from_here_data(
+                        self.coordinator.data, "windSpeed", offset
+                    ),
+                    ATTR_FORECAST_WIND_BEARING: get_attribute_from_here_data(
+                        self.coordinator.data, "windDirection", offset
+                    ),
+                    ATTR_FORECAST_CONDITION: get_condition_from_here_data(
+                        self.coordinator.data, offset
+                    ),
+                }
+            )
+        return data
+
+    @property
+    def entity_registry_enabled_default(self):
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return self._mode == DEFAULT_MODE
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+
+        return {
+            "identifiers": {(DOMAIN, self._name)},
+            "name": self._name,
+            "manufacturer": "here.com",
+        }
+
+
+def get_condition_from_here_data(here_data: list, offset: int = 0) -> str | None:
+    """Return the condition from here_data."""
+    try:
+        return [
+            k
+            for k, v in CONDITION_CLASSES.items()
+            if get_attribute_from_here_data(here_data, "iconName", offset) in v
+        ][0]
```

Comment:
Since we only want the first item we could use a generator expression and `next` to get it. It's faster. It also allows a default value on missing data.
```py
return next((k for k, v in ...), None)
```

Reply:
Implemented in https://github.com/home-assistant/core/pull/28910/commits/8c3d47d60a0677c49cc0bc32941588f6a735f496